manim/config.py

```diff
@@ -239,3 +239,45 @@ def get_camera_configuration(args):
         camera_config["background_opacity"] = 0
 
     return camera_config
+
+
+def initialize_directories(cfgargs):
+    dir_config = {}
+    dir_config["media_dir"] = cfgargs["media_dir"] or dirs.MEDIA_DIR
+    dir_config["video_dir"] = cfgargs["video_dir"] or dirs.VIDEO_DIR
+
+    if not (cfgargs["video_dir"] and cfgargs["tex_dir"]):
+        if cfgargs["media_dir"]:
+            if not os.path.isdir(dir_config["media_dir"]):
+                os.makedirs(dir_config["media_dir"])
+        if not os.path.isdir(dir_config["media_dir"]):
+            dir_config["media_dir"] = "./media"
+        else:
+            print(
+                f"Media will be written to {dir_config['media_dir'] + os.sep}. You can change "
+                "this behavior with the --media_dir flag, or by adjusting dirs.py.,"
+            )
+    else:
+        if cfgargs["media_dir"]:
+            print(
+                "Ignoring --media_dir, since both --tex_dir and --video_dir were passed."
+            )
+
+    dir_config["tex_dir"] = (cfgargs["tex_dir"]
+                             or dirs.TEX_DIR
+                             or os.path.join(dir_config["media_dir"], "Tex"))
+    dir_config["text_dir"] = (cfgargs["text_dir"]
+                              or dirs.TEXT_DIR
+                              or os.path.join(dir_config["media_dir"], "texts"))
+
+    if not cfgargs["video_dir"] or dirs.VIDEO_DIR:
+        dir_config["video_dir"] = os.path.join(dir_config["media_dir"], "videos")
+        
+    for folder in [dir_config["video_dir"], dir_config["tex_dir"], dir_config["text_dir"]]:
+        if folder != "" and not os.path.exists(folder):
+            os.makedirs(folder)
+
+    dirs.MEDIA_DIR = dir_config["media_dir"]
+    dirs.VIDEO_DIR = dir_config["video_dir"]
+    dirs.TEX_DIR = dir_config["tex_dir"]
+    dirs.TEXT_DIR = dir_config["text_dir"]
```

Comment:
This should include and `IMAGE_DIR` and `TEX_DIR` and `TEXT_DIR` should be combined into `SVG_DIR`. The format we discussed was
```
media
  <module name>
    <scene name>
      svg
      images
      video
        <resolution>p<framerate>
```

Reply:
Point taken and I agree. Not going to do that in this PR though. All clean this up, and then make the folder structure a separate issue and PR.